src/steps/trackTransaction.ts

```diff
@@ -91,7 +91,8 @@ export function trackTransaction(
     //   console.log('skipping first step')
     //   continue
     // }
-    if (iters > 20) {
+    const uiEvent = tr.getMeta('uiEvent')
+    if (iters > 20 && uiEvent != 'cut') {
```

Comment:
it would be ideal to still prevent infinite loop somehow even in case of cutting. Even let's say if its a couple of hundred transacations but still a finite number